adept/_base_.py

```diff
@@ -386,7 +387,8 @@ def val_and_grad(self, modules: Dict = None) -> Tuple[float, Dict, Tuple[Solutio
             t0 = time.time()
             with tempfile.TemporaryDirectory(dir=self.base_tempdir) as td:
                 post_processing_output = self.adept_module.post_process(run_output, td)
-                robust_log_artifacts(td)  # logs the temporary directory to mlflow
+                if export:
+                    robust_log_artifacts(td)  # logs the temporary directory to mlflow
                 if "metrics" in post_processing_output:
                     mlflow.log_metrics(post_processing_output["metrics"])
```

Comment:
```suggestion
            post_processing_output = self.adept_module.post_process(run_output)
            if export_all:
                with tempfile.TemporaryDirectory(dir=self.base_tempdir) as td:
                    robust_log_artifacts(td)  # logs the temporary directory to mlflow
            if export_metrics and "metrics" in post_processing_output:
                    mlflow.log_metrics(post_processing_output["metrics"])
```
Personally, I'd find it more helpful if there was more customisation on what was exported. Metrics may be data-intensive and undesirable for logging. Sometimes, Mlflow might want to be removed entirely including on setup. We could have multiple flags `export_all, export_metrics, export_init, disable_mlflow`.

I think `adept_module.post_process` should also be refactored so that it no longer takes `td` as an input, forcing the logging to happen on a separate line. This may be a breaking change but I know that `lagradept` for example does not actually use the `td` argument.

Nevertheless, happy for you to merge this to use for now and we work in something more all encompassing afterwards.

Reply:
Saw you said something very similar in your last comment, that approach sounds great!

Reply:
also should just be called something else, it really is only a logging wrapper